basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+
+            }
+
+            //Create Trajectory geometry
+            double delta_step = 1.0 / start_centerline.size();
+
+            for (int i = 0; i < start_centerline.size(); i++)
+            {
+                lanelet::BasicPoint2d current_position;
+                lanelet::BasicPoint2d start_lane_pt = start_centerline[i];
+                lanelet::BasicPoint2d end_lane_pt = end_centerline[i];
+                double delta = delta_step * i;
+                current_position.x() = end_lane_pt.x() * delta + (1 - delta) * start_lane_pt.x();
+                current_position.y() = end_lane_pt.y() * delta + (1 - delta) * start_lane_pt.y();
+
+                centerline_points.push_back(current_position);
+            }
+
+            //Add points from following lanelet to provide sufficient distance for adding buffer
+            auto following_lanelets = wm->getMapRoutingGraph()->following(ending_lanelet, false);
+            if(!following_lanelets.empty()){
+            //Arbitrarily choosing first following lanelet for buffer since points are only being used to fit spline
+            auto following_lanelet_centerline = following_lanelets.front().centerline2d().basicLineString();
+            centerline_points.insert(centerline_points.end(), following_lanelet_centerline.begin(), 
+                                                                        following_lanelet_centerline.end());
+            }
+
+            return centerline_points;
+        }
+
+        std::vector<std::vector<lanelet::BasicPoint2d>> resample_linestrings_to_same_length(std::vector<lanelet::BasicPoint2d> line_1, std::vector<lanelet::BasicPoint2d> line_2){
+            std::vector<std::vector<lanelet::BasicPoint2d>> output;
+            
+            //Fit centerlines to a spline
+            std::unique_ptr<smoothing::SplineI> fit_curve_1 = compute_fit(line_1); // Compute splines based on curve points
+            if (!fit_curve_1)
+            {
+                throw std::invalid_argument("Could not fit a spline curve along the starting_lane centerline points!");
+            }
+
+            std::unique_ptr<smoothing::SplineI> fit_curve_2 = compute_fit(line_2); // Compute splines based on curve points
+            if (!fit_curve_2)
+            {
+                throw std::invalid_argument("Could not fit a spline curve along the ending_lane centerline points!");
+            }
+
+            //Sample spline to get centerlines of equal size
+            std::vector<lanelet::BasicPoint2d> all_sampling_points_line1;
+            std::vector<lanelet::BasicPoint2d> all_sampling_points_line2;
+
+            size_t total_point_size = std::min(line_1.size(), line_2.size());
+
+            all_sampling_points_line1.reserve(1 + total_point_size * 2);
+            std::vector<double> downtracks_raw_line1 = carma_wm::geometry::compute_arc_lengths(line_1);
+            int total_step_along_curve1 = static_cast<int>(downtracks_raw_line1.back() / 2.0);
```

Comment:
Also if we're accessing `*.back()` it's probably worth documenting the assumption that the input lists are non-empty and/or actually checking and enforcing that ourselves. We should do what we can to ensure that bad input doesn't segfault nodes but is instead captured and logged, then handled.

Reply:
Still think it would be a good idea to check at the top of this method to catch if either of the input lists are empty and gracefully fail. We've seen several cases where 0 length maneuvers get generated via some weird geometry or other edge cases, and having a proper exception for those cases is substantially easier to trace and debug than a silent segfault or other form of undefined behavior.

In general validating inputs like that as a form of defensive programming is something I'd advocate for, if a method depends on a particular property of it's inputs (non-emptiness, non-nullity, etc.) it should validate those properties itself so that errors can be caught. If most or all functions do this then it means that errors can be caught (and hopefully handled) closer to their actual origin instead of deeper in the callstack.

Reply:
For this one, there is a curve fitting that happens right before this which requires there be atleast 4 points in the vector. There is an exception being thrown if that fails, would that be a sufficient check?